### PR TITLE
Add an optional presubmit job for kwok performance tests

### PR DESCRIPTION
**Pull request by Sreeja1725**
**What this PR does / why we need it**: Added an optional presubmit job to test against https://github.com/kubevirt/kubevirt/pull/12117

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -2066,3 +2066,67 @@ presubmits:
...
+      testgrid-dashboards: kubevirt-presubmits
+    cluster: kubevirt-prow-workloads
+    decorate: true
+    decoration_config:
+      grace_period: 5m0s
+      timeout: 4h0m0s
+    extra_refs:
+    - base_ref: main
+      org: kubevirt
+      repo: kubevirt
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2081]: This extra ref should not be needed here.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 2]
```diff
@@ -2066,3 +2066,63 @@ presubmits:
...
+          # Build the perfscale-audit binary
+          make bazel-build
+          # Create k8s cluster
+          make cluster-up
+          # Push and deploy kubevirt
+          make cluster-sync
+          FUNC_TEST_ARGS="--no-color --seed=42" make kwok-perftest
+        env:
+        - name: KUBEVIRT_PROVIDER
+          value: k8s-1.29
```
Comment (xpivarc) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2106]: Can we use latest?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 3]
```diff
@@ -2066,3 +2066,63 @@ presubmits:
...
+        - name: KUBEVIRT_STORAGE
+          value: hpp
```
Comment (xpivarc) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2108]: Do we actually need storage?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 4]
```diff
@@ -2066,3 +2066,63 @@ presubmits:
...
+        - name: KUBEVIRT_MEMORY_SIZE
+          value: 10G
+        - name: KUBEVIRT_NUM_NODES
+          value: "4"
```
Comment (xpivarc) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2112]: Do we need so many workers if we are just simulating the workers?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 2 above)
Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2106]: Thanks that makes sense

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 4 above)
Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2112]: @Sreeja1725 :ping_pong:

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 3 above)
Comment (Sreeja1725) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2108]: @xpivarc I believe logs and artifacts are stored in KUBEVIRT_STORAGE. Is this storage configuration specifically for storage within the cluster? The logs appear to be on the pod executing the test, which is external to the cluster, are these logs not being stored?

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 4 above)
Comment (Sreeja1725) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2112]: Removed the  KUBEVIRT_NUM_NODES to use the default num of nodes, by default kubevirt is using 2 workers to run platform nodes.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml (hunk 3 above)
Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2108]: No, the `KUBEVIRT_STORAGE` is for in cluster storage. Logs as objects,logs etc are collected and pushed to bucket. @rthallisey @alaypatel07 do we need storage for anything?

Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2108]: Let's drop this, we merge and try in kubevirt. We can come back if something fails...

Comment (alaypatel07) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:2108]: I am not sure, @Sreeja1725 we can start with removing the storage and merge the PR, we can change it in future if needed.

Comment (Sreeja1725) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml]: Removed the KUBEVIRT_STORAGE for now

**Review by xpivarc — COMMENTED** [review]
/approve
@brianmcarey PTAL

**Review by brianmcarey — COMMENTED** [review]
/lgtm

**Pull request merged**